server/src/main/java/org/elasticsearch/index/codec/tsdb/es819/ES819TSDBDocValuesConsumer.java

```diff
@@ -121,10 +122,15 @@ public SortedNumericDocValues getSortedNumeric(FieldInfo field) throws IOExcepti
             writeSkipIndex(field, producer);
         }
 
-        writeField(field, producer, -1);
+        writeField(field, producer, -1, null);
     }
 
-    private long[] writeField(FieldInfo field, TsdbDocValuesProducer valuesProducer, long maxOrd) throws IOException {
+    private long[] writeField(
+        FieldInfo field,
+        TsdbDocValuesProducer valuesProducer,
+        long maxOrd,
+        CheckedIntConsumer<IOException> docCountConsumer
```

Comment:
nit: maybe just pass down the `OffsetsAccumulator` here? No need for an int consumer abstraction at this level.